src/main/java/gov/usgs/wma/waterdata/collections/CollectionsController.java

```diff
@@ -129,19 +136,59 @@ public String getOgcCollectionFeatures(
 							content=@Content())
 			},
 			externalDocs=@ExternalDocumentation(url="http://docs.opengeospatial.org/is/17-069r3/17-069r3.html#_feature_")
-			)
+		)
 	@GetMapping(value = "collections/{collectionId}/items/{featureId}", produces = MediaType.APPLICATION_JSON_VALUE)
 	public String getOgcCollectionFeature(
 			@RequestParam(value = "f", required = false, defaultValue = "json") String mimeType,
 			@PathVariable(value = "collectionId") String collectionId,
 			@PathVariable(value = "featureId") String featureId, HttpServletResponse response) {
 
-		String rtn = collectionsDao.getCollectionFeatureJson(collectionsParams.buildParams(collectionId, featureId));
-		if (rtn == null) {
-			response.setStatus(HttpStatus.NOT_FOUND.value());
-		}
+		return resultOr404(response, 
+				collectionsDao.getCollectionFeatureJson(collectionsParams.buildParams(collectionId, featureId)));
+	}
 
-		return rtn;
+	@Operation(
+			description="Return GeoJSON Data specific to the requested Monitoring Location and Time Series.",
+			responses= {
+					@ApiResponse(
+							responseCode="200",
+							description="GeoJSON representation of the Time Series.",
+							content=@Content(schema=@Schema(implementation=TimeSeriesGeoJSON.class))),
+					@ApiResponse(
+							responseCode="404",
+							description="The requested Time Series was not found.",
+							content=@Content())
+			},
+			externalDocs=@ExternalDocumentation(url="https://github.com/opengeospatial/omsf-profile/tree/master/omsf-json")
+		)
+	@GetMapping(value="collections/{collectionId}/items/{featureId}/time-series/{timeSeriesId}", produces=MediaType.APPLICATION_JSON_VALUE)
+	public String getTimeSeries(
+			@PathVariable(value="collectionId") String collectionId, // ex: networkId,
+			@PathVariable(value="featureId") String featureId, // ex: monitoringLocationId
+			@PathVariable(value="timeSeriesId") String timeSeriesId, //ex: USGS-123456
+			HttpServletResponse response) {
+		
+		// verify the collection and feature exist before fetching the time series
+		return resultOr404(response, timeSeriesDao.getTimeSeries(collectionId, featureId, timeSeriesId));
 	}
 
+	
+	/**
+	 * Helper method to set the response to 404 if there is no result from the request.
+	 * @param lambda an action to do to respond to the request that returns a string upon success
+	 * @param result the string response from any request
+	 * @return on a successful response it will be the string provided by the lambda.
+	 */
+	protected String resultOr404(HttpServletResponse response, String result) {
+		// Ensure the response code it set only once. Some impls might prevent multiple settings.
+		// Testing revealed that the response code is preset to 200 (not null)
+		if (response.getStatus() == HttpStatus.NOT_FOUND.value()) {
```

Comment:
Is this necessary? We really shouldn't get here with the result status set. To say it differently, once the status is set, we should not continue processing.

Reply:
I only added it because you were concerned about double setting. I think this is a lot less confusing than you are making it out to be --- but, as I said in a prior comment "above", I will take another look at it. To me the copy-paste is worse. This method encapsulates the common need.

Reply:
Sorry for not being clear, I though the previous logic (calling the getOgcCollections method to validate the collection feature) could have tried to set the status twice.

In my opinion, setting the 404 based on not getting data back from the dao could be considered copy-paste. Setting a 404 based on input parameter validation is not a copy-paste with that.

Reply:
ok